### PR TITLE
Documentation Refresh (close #1050)

### LINKED ISSUE #1050: HTML Elements Documentation Structure
I'd like to clean up the documentation surrounding HTML elements. 

I'm proposing that:

- [ ] HTML elements are split into subcategories of "decorative" (e.g. `pl_code`, `pl_figure`, ...) and "input fields" (e.g. `pl_number_input`, `pl_multiple_choice`, ... ) 
- [ ] Renaming and tagging test suite questions to match more closely to what they show.
- [ ] Adding links in the documentation to view source of test question code in the repository (e.g. `question.html`, `info.json`, ... )

### PR DESCRIPTION
**Pull request by coatless**
This PR is a spiritual successor to PR #1069 and closes #1050.

Contained within the PR:

- New introduction to element documentation
- Each element documentation entry is standardardized with a picture under `/doc/elements/pl-<element>.png` 
- Reorganized the PL example gallery

Documentation for each element is organized under:

```md
## `pl-element-name` element

Short Description

#### Sample Element

![Element picture](elements/pic.png)

Customize the `pl-element-name` element with:

#### Customizations

Markdown table (unformatted)

#### Detail

Custom notes related to `server.py` or additional requirements

#### Example implementations

- Link to GitHub source of question
- Add Magic Link once activated

#### See also

References to other portions of the documentation such as:

- other submission elements
- related decoration elements (e.g. matrix outputs)
- best practices
```

Sample:

Screen Shot 2019-03-26 at 1 45 20 PM

### CODE REVIEW

File: doc/elements.md [hunk 1]
```diff
@@ -1,16 +1,89 @@
...
+images, files, and code display. The following **decorative** elements are available:
+
+- [`pl-code`](#pl-code-element): Displays code rendered with the appropriate
+  syntax highlighting. 
+- [`pl-figure`](#pl-figure-element): Embed an image file in the question.
+- [`pl-file-download`](#pl-file-download-element): Enable file downloads for
+  data-centric questions.
+- [`pl-matrix-output`](#pl-matrix-output-element): Displays dynamically
+  generated matrices.
+- [`pl-matrix-latex`](#pl-matrix-latex-element): Displays dynamically
```
Comment (mwest1066) [inline doc/elements.md]: Same description for these the above two elements.

File: doc/elements.md [hunk 2]
```diff
@@ -1,16 +1,89 @@
...
+  such as -71, 0, 5, 21, and so on.
+- [`pl-symbolic-input`](#pl-symbolic-input-element): Fill in a **symbolic** value
+  such as `x^2`, `sin(z)`, `mc^2`, and so on.
+- [`pl-string-input`](#pl-string-input-element): Fill in a **string** value
+  such as "Illinois", "GATTACA", "computer", and so on.
+- [`pl-matrix-component-input`](#pl-matrix-component-input-element): Fill in
+  a **matrix** using grid that has an input area for each element.
+- [`pl-matrix-input`](#pl-matrix-input-element): Supply a matrix in a supported
+  programming language format.
+- [`pl-file-editor`](#pl-file-editor-element): Provide an inbrowser code editor
```
Comment (mwest1066) [inline doc/elements.md]: Hyphenate `in-browser`?

File: doc/elements.md [hunk 3]
```diff
@@ -1,16 +1,89 @@
...
+  syntax highlighting. 
+- [`pl-figure`](#pl-figure-element): Embed an image file in the question.
+- [`pl-file-download`](#pl-file-download-element): Enable file downloads for
+  data-centric questions.
+- [`pl-matrix-output`](#pl-matrix-output-element): Displays dynamically
+  generated matrices.
+- [`pl-matrix-latex`](#pl-matrix-latex-element): Displays dynamically
+  generated matrices. 
+- [`pl-question-panel`](#pl-question-panel-element): Displays the text of a question
+- [`pl-submission-panel`](#pl-submission-panel-element): Displays the inputted
```
Comment (mwest1066) [inline doc/elements.md]: Maybe just `input answers`?

File: doc/elements.md [hunk 4]
```diff
@@ -1,16 +1,89 @@
...
+  answers of the student.
+- [`pl-answer-panel`](#pl-answer-panel-element): Displays the correct
+  answer to a given question.
+- [`pl-external-grader-results`](#pl-external-grader-results-element):
+  Displays results from questions that are externally graded.
+- [`pl-variable-score`](#pl-variable-score-element): Displays a partial score
```
Comment (mwest1066) [inline doc/elements.md]: Do we actually need this anymore? I think this is just a legacy element for v2 questions. I believe that all v3 elements do partial score display automatically.

Comment (coatless) [reply doc/elements.md]: Legacy at the moment. I'm fine with removing if need be.

Comment (mwest1066) [reply doc/elements.md]: Yeah, let’s drop `pl-variable-score` from the docs. Or maybe add a `deprecated` section and say that these elements still exist, but they should not be used in new questions and support for them may be removed in the future?

Comment (coatless) [reply doc/elements.md]: Added a deprecated section.

File: doc/elements.md [hunk 5]
```diff
@@ -1,16 +1,93 @@
...
+- [`pl-code`](#pl-code-element): Displays code rendered with the appropriate
+  syntax highlighting. 
+- [`pl-figure`](#pl-figure-element): Embed an image file in the question.
+- [`pl-file-download`](#pl-file-download-element): Enable file downloads for
+  data-centric questions.
+- [`pl-variable-output`](#pl-variable-output-element): Displays matrices in 
+  code form for supported programming languages.
+- [`pl-matrix-latex`](#pl-matrix-latex-element): Displays matrices using 
+  appropriate LaTeX commands for use in a mathematical expression.
+- [`pl-question-panel`](#pl-question-panel-element): Displays the text of a question.
```
Comment (nwalters512) [inline doc/elements.md]: Consider moving these to a **conditional** section? They don't so much display the text of a question as they do conditionally render their children depending on panel; I think that distinction is worth making clear and expanding on.

File: doc/elements.md [hunk 6]
```diff
@@ -1,16 +1,93 @@
...
+  figure.
+
+Note: PrairieLearn Elements listed next have been **deprecated**. These elements
+will be removed at a future date.
+
+- [`pl-variable-score`](#pl-variable-score-element): Displays a partial score
+  for a submitted element.
+    - **Deprecated** as submission elements in `v3` all have score display options.
+
+## Submission Elements --
```
Comment (nwalters512) [inline doc/elements.md]: The double dash after this isn't really adding much and can probably be removed.

File: doc/elements.md (hunk 5 above)
Comment (coatless) [reply doc/elements.md]: Agreed. I've rearranged the document to account for this.

Though, I think another round (not this PR) should probably add a section to the documentation that emphasizes how one can use these w.r.t custom elements.

File: doc/elements.md (hunk 6 above)
Comment (coatless) [reply doc/elements.md]: Done.

**Pull request merged**